src/stash/src/persist.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::collections::{hash_map, HashMap};
+use std::fmt;
+use std::hash::Hash;
+use std::vec;
+
+use ore::metrics::MetricsRegistry;
+use persist::client::{RuntimeClient, StreamWriteHandle};
+use persist::indexed::Snapshot;
+use persist::runtime::{self, RuntimeConfig};
+use persist::storage::{Blob, Log};
+use persist_types::Codec;
+use timely::progress::Antichain;
+
+use crate::{Stash, StashError, StashOp};
+
+type Error = StashError<persist::error::Error>;
+
+/// A stash backed by a [`persist`] collection.
+#[derive(Debug)]
+pub struct PersistStash<K, V> {
+    _client: RuntimeClient,
+    write_handle: StreamWriteHandle<K, V>,
+    data: HashMap<K, V>,
```

Comment:
I think this field likely deserves a bit of commentary about why it's here, what it's doing, etc

am I correct that if/when persist actually exposed structured access to the stored data, that we could replace this with that?